Jasp results summary stats

ASarafoglou opened the pull request:
Rewrote Analyses in Summary Stats Module to JASP results (except Bayesian Correlation Pairs).

Commit:
rewrite to JASP results

modified .qml and R file. No unit tests available yet.

added unit tests

added unit tests

integrate jasp results for independent samples ttest

rewrite on Inpedendent samples t-test and binomial test

made sure that structure in both analyses are the same.
Commented out all plots for independent samples t-test, because functions will be provided later.

include: one sample t-test

one samples t-test written in JaspResults
this analysis excludes all plots

added paired samples t-test to JaspResults

including unit test for main table
excluding prior and posterior plots

bayesian correlation pairs in jaso results

analysis shows main table
excluding prior and posterior plot and robustness plot

replace correlation coefficient in results table

correlationResults$r refers only to pearsons rho
correlationResults$tauObs refers only to kendalls tau

correlationResults$stat refers to the test value specified by user

added unit test for main table in bayesian correlation pairs

linear regression bayesian to jaspResults

added bayesian linear regression to jaspResults
did not inlcude robustness check plot
did include unit tests

Commit:
rewrite on the summarystatsbinomialtestbayesian

analysis runs now in Qt
- Bayes factor wheel does not show when 'additional info' is ticked
everything else seems to work now

Commit:
started rewrite of Bayesian One Sample TTest

- some minor changes in bayesian binomial
- started rewrite in one sample t-test (test is now returning table and prior/posterior plot)
- added prior/posterior plot functionality in one sample t-test
- added robustness plot (however, not functional yet)

Commit:
renaming function in summary stats binomial test

Commit:
Rewrite one sample t-test

rewrote analysis.
Moved 'inferential plot', 'robustness plot' and 'prior posterior plot' to common file. Robustness plot is working now.

Commit:
summary stats stuff

Commit:
fix plotting dependencies

Commit:
finished up bayesian binomial test

added unit tests
added error checks

Commit:
delete BF subscripts from summary stats binomial

BF subscripts no longer needed for prior and posterior plot

vandenman committed:
rewrote linear regression and added unit tests

Commit:
added plots for one sample t-test

added prior/posterior plot
deleted old plotting functions

Commit:
remove browser statement

Commit:
solved some state issues

Commit:
fixed dependencies for plots, paried = TRUE for all BF computations

included dependencies for prior/posterior plot and robustness plot
turned on paired = TRUE for all bf computations in one sample case

Commit:
added unit test for one sample t test

Commit:
started rewrite of paired sampled t-test

Commit:
rewrote paired samples ttest and added unit tests

- also: merged common functions with one samples ttest

ASarafoglou committed:
rewrite independent samples t-test and made general ttest functions

added unit tests

ASarafoglou committed:
some style improvements

ASarafoglou committed:
revert correlation changes

TimKDJ commented on Resources/Summary Statistics/qml/SummaryStatsBinomialTestBayesian.qml:
```diff
@@ -24,7 +24,7 @@ import JASP.Widgets 1.0
 Form 
 {
-	usesJaspResults: false
+	usesJaspResults: true
```
You can just omit `usesJaspResults: true` -- this is the default

ASarafoglou replied:
I deleted this line.

TimKDJ commented on JASP-Engine/JASP/R/summarystatsbinomialtestbayesian.R:
```diff
@@ -15,348 +15,269 @@
+  .checkErrors.summarystats.binomial(options)
+  
+  # Compute the results and create main results table
+  summaryStatsBinomialResults <- .summaryStatsBinomialMainFunction(jaspResults, options)
```
The inconsistency in function names (`.checkErrors.summarystats.ttest` <-> `.summaryStatsTTestMainFunction`) makes me a little sad

ASarafoglou replied:
Nobody wants to see you being sad :(
I fixed this for .checkErrors.summarystats.ttest() and .hypothesisType.summarystats.binomial()

(will do this for the other summary stats analyses as well)

TimKDJ commented on JASP-Engine/JASP/R/summarystatsbinomialtestbayesian.R:
```diff
@@ -15,348 +15,269 @@
+    jaspResults[["binomialContainer"]] <- createJaspContainer()
+    jaspResults[["binomialContainer"]]$dependOn(c("successes", "failures", "betaPriorParamA", "betaPriorParamB", "testValue", "hypothesis"))
+  }
+  
```
Suggestion:

If you add
`container <- jaspResults[["binomialContainer"]]`
here, then you can use `container` instead of `jaspResults[["binomialContainer"]]` and your function becomes quite a bit more legible

TimKDJ commented on JASP-Engine/JASP/R/summarystatsbinomialtestbayesian.R:
```diff
@@ -15,348 +15,269 @@
+  bayesianBinomialTable$addColumnInfo(name = "failures" , title = "Failures"  , type = "integer")
+  bayesianBinomialTable$addColumnInfo(name = "theta0"   , title = "Test value", type = "number")
+  bayesianBinomialTable$addColumnInfo(name = "BF"       , title = bfTitle     , type = "number")
+  bayesianBinomialTable$addColumnInfo(name = "pValue"   , title = "p"         , type = "number")
```
```suggestion
  bayesianBinomialTable$addColumnInfo(name = "pValue"   , title = "p"         , type = "pvalue")
```

ASarafoglou replied:
Corrected.

TimKDJ commented on JASP-Engine/JASP/R/summarystatsregressionlinearbayesian.R:
```diff
@@ -15,1294 +15,327 @@
+  }
+  
+  # check if number of covariates is correct in H0
+  if(options$numberOfCovariatesNull==0 && options$unadjustedRSquaredNull==0) {
```
This if statement is in direct contrast with the one it encapsulates. I think it can be deleted altogether?

ASarafoglou replied:
You are right. I corrected it.

TimKDJ commented on JASP-Engine/JASP/R/commonsummarystatsttestbayesian.R:
```diff
@@ -157,1709 +77,559 @@
+  # check whether % error could be computed
+  if(is.na(ttestTableData$error) || is.null(ttestTableData$error)){
+    ttestTableData$error <- NaN
+    ttestTableMessage    <- "Proportional error estimate could not be computed."
```
This should be a column (or cell, whichever, it only has 1 cell anyway) specific footnote, rather than a general table footnote.

ASarafoglou replied:
Fixed.

TimKDJ reviewed (changes requested):
I think you did a good job :)
Do have a few comments though:

**Binomial:**
- [ ] The additional info of the Prior posterior plot no longer takes into account what the "Bayes Factor"/"Alt. Hypothesis" is. It should change if you go from `BF10` to `BF01`, and also if you go from `> Test value` to `< Test value`
 
---

**Regression:**
- [ ] BF results are way different, but error term is the same:
<img width="2541" alt="Screenshot 2019-10-31 at 11 06 13" src="https://user-images.githubusercontent.com/18737241/67943735-044bd900-fbdb-11e9-82f3-d62c4c5e1057.png">

- [ ] User prior (gray dot) is not on the line:
<img width="568" alt="Screenshot 2019-10-31 at 11 07 38" src="https://user-images.githubusercontent.com/18737241/67943765-13cb2200-fbdb-11e9-9323-d750fec51125.png">

---

**ttest independent samples:**
- [ ] The Bayes factor robustness check plot does not change when you change the type of the "Bayes Factor"

- [ ] Results seem incorrect for informed priors (they appear to be the correct for the other ttests)
<img width="2534" alt="Screenshot 2019-10-31 at 11 59 15" src="https://user-images.githubusercontent.com/18737241/67944269-580af200-fbdc-11e9-9ebb-9b36749b64e9.png">

TimKDJ commented:
Btw, please fix the code comments in all files (if I made a comment about something in file A I did not make the same comment again in file B).

TimKDJ commented:
@ASarafoglou how is the updated code coming along?

ASarafoglou commented:
I post my progress by updating this comment:

 **Binomial:**
 
 * [x]  The additional info of the Prior posterior plot no longer takes into account what the "Bayes Factor"/"Alt. Hypothesis" is. It should change if you go from `BF10` to `BF01`, and also if you go from `> Test value` to `< Test value`
 
**Regression:** 
* [x]  BF results are way different, but error term is the same:
* [x]  User prior (gray dot) is not on the line.
        Corrected. Also the robustness plot additional info changes now with the BF type.

 **ttest independent samples:**
 
* [x]  The Bayes factor robustness check plot does not change when you change the type of the "Bayes Factor"
* [x]  Results seem incorrect for informed priors (they appear to be the correct for the other ttests)

ASarafoglou committed:
fixed additional info in binomial test and improved coding style

ASarafoglou committed:
corrected BF computation in Linear Regression

and style improvements

ASarafoglou committed:
Style improvements TTest and fixed dependency

ASarafoglou committed:
fixed bayes factor computation for independent samples ttest

ASarafoglou committed:
general coding style improvements

ASarafoglou commented:
@TimKDJ I think I am done with the rewrite..??

![giphy](https://user-images.githubusercontent.com/20499997/68873455-f8463800-06ff-11ea-8819-9d4795f21a29.gif)

TimKDJ commented:
SumStats correlation expects the '.getBayesfactorTitle.summarystats.ttest' function (see [travis](https://travis-ci.org/jasp-stats/jasp-desktop/builds/611939073?utm_source=github_status&utm_medium=notification)), could you add this to that R file? At least until Lexi's rewrite is finished

ASarafoglou committed:
get Bayes factor title for correlation pairs

TimKDJ commented on JASP-Tests/R/tests/testthat/test-summarystatscorrelationbayesianpairs.R:
```diff
@@ -1,29 +1,16 @@
-context("SummaryStatsCorrelationBayesianPairs")
+context("Summary Statistics Bayesian Correlation Pairs")
```
Can you undo the changes you made to this file?
This is what fails the unit tests right now and considering you didn't rewrite correlation these changes seem unnecessary.

ASarafoglou replied:
corrected.

TimKDJ reviewed (changes requested):
The changes you made look good (mostly)! Did notice the following though:
Binomial:
- [ ] Ticking “Prior and posterior” crashes the analysis: Error in .summaryStatsBinomialMainFunction(jaspResults, options): object 'container' not found

Regression:
- [ ] If you create a bayes factor robustness check plot, then it says “Cauchy prior width” on the x-axis rather than “r scale”
- [ ] The user prior in the additional info portion of the plot appears to be incorrect (related to the gray dot not being on the line, which is also the case here).

TTest independent samples:
- [ ] Same as binomial: Ticking “Prior and posterior” crashes the analysis: Error in .summaryStatsBinomialMainFunction(jaspResults, options): object 'container' not found

All of these points can be found in [rewriteSumStats.jasp.zip](https://github.com/jasp-stats/jasp-desktop/files/3869016/rewriteSumStats.jasp.zip)

ASarafoglou committed:
undo test file changes for bayesian correlation

- undo changes in test file for bayesian correlation pairs

ASarafoglou committed:
fixed additional info in BF robustness plot (linear regression)

also changed x axis title

ASarafoglou committed:
fixed container issue

fixed container issue
resolved additional info bug in robustness plot (summary stats t-test); max BF is now displayed correctly

ASarafoglou commented:
Binomial:

* [x]  Ticking “Prior and posterior” crashes the analysis: Error in .summaryStatsBinomialMainFunction(jaspResults, options): object 'container' not found
 
Regression:

* [x]  If you create a bayes factor robustness check plot, then it says “Cauchy prior width” on the x-axis rather than “r scale”
* [x]  The max BF in the additional info portion of the plot appears to be incorrect (related to the gray dot not being on the line, which is also the case here).

@TimKDJ I don't see that the user prior is incorrect. Can you give an example where this occurs? I just saw that the max BF01 was actually not the correct number, so I fixed that instead (for both the t-tests and the linear regression)

TTest independent samples:
 
* [x]  Same as binomial: Ticking “Prior and posterior” crashes the analysis: Error in .summaryStatsBinomialMainFunction(jaspResults, options): object 'container' not found

ASarafoglou committed:
fixed the user prior issue in bayesian linear regression

grey dots are now always on the line of the robustness plot
deleted wrong BF list from the code

TimKDJ reviewed (changes requested):
Sorry for my horribly late response!
Just small things now ;)

Independent Samples TTest
- [ ] Robustness plot should have no dots on the line if there is no legend (like it used to be previously)

- [ ] The piechart on top of the prior posterior plot does not look ok (seems to be caused by the next comment):
<img width="1243" alt="Screenshot 2019-12-10 at 09 48 48" src="https://user-images.githubusercontent.com/18737241/70517887-a5627380-1b39-11ea-8967-9ce40d760c6f.png">

- [ ] The bayes factor in the prior posterior plot is whatever type of Bayes Factor is selected (e.g., BF10 or Log(BF10)), but this is 1) not reflected in the label — it should say Log(BF) if that’s what we’re showing; and 2) the value does not change when you change the Bayes Factor type after you create the plot. I think this should never be the log value though?

---
Paired Samples TTest
- [ ] The label of the robustness plot says log(BF0+) but I don’t think that’s what it’s actually showing:
old <-> new
<img width="1828" alt="Screenshot 2019-12-10 at 10 00 18" src="https://user-images.githubusercontent.com/18737241/70517747-603e4180-1b39-11ea-8592-225e7211b6a1.png">

---
Binomial
- [ ] Space between “Population proportion” and “theta”
<img width="596" alt="Screenshot 2019-12-10 at 10 34 06" src="https://user-images.githubusercontent.com/18737241/70517716-4ef53500-1b39-11ea-89cc-dcf6f35d7829.png">

ASarafoglou committed:
fixed robustness test in sumstats t-test; fixed binomial x-axis

t-tests:
- main container now depends on BFtype (easiest solution for the plots)
- plots do not show logBF anymore
- robustness plots does not show dots when additional info is not selected
- unit test plots changed

binomial:
- fixed the x-axis

ASarafoglou commented:
Independent Samples TTest:

- [x] Robustness plot should have no dots on the line if there is no legend (like it used to be previously)

- [x] The piechart on top of the prior posterior plot does not look ok (seems to be caused by the next comment)

- [x] The Bayes factor in the prior posterior plot is whatever type of Bayes Factor is selected (e.g., BF10 or Log(BF10)), but this is 1) not reflected in the label — it should say Log(BF) if that’s what we’re showing; and 2) the value does not change when you change the Bayes Factor type after you create the plot. I think this should never be the log value though?

The Label is not adapting and I don't allow log Bayes factors in the prior posterior plot.

Paired Samples TTest:
- [x] The label of the robustness plot says log(BF0+) but I don’t think that’s what it’s actually showing:
old <-> new

Binomial:

- [x] Space between “Population proportion” and “theta”

ASarafoglou committed:
added setSeedJASP in common file

TimKDJ commented:
Here comes the.. um.. merge shark!
![giphy](https://pa1.narvii.com/6026/72631d306c2048c49e58d637982d0f5b6805ffa4_hq.gif)

ASarafoglou commented:
Hahaha amazing! xD
